### DISCUSSION TITLE
Looking for detailed documentation around zInfo

### DISCUSSION THREAD

**Discussion by aochagavia** [awslabs/soci-snapshotter#1117] [Q&A]
I'm interested in writing an independent implementation of SOCI in Rust, specifically meant to _create_ SOCI Indexes. The TOC part seems easy to do (it looks somewhat similar to the eStargz TOC), but the specific details on zInfo aren't covered by the [docs](https://github.com/awslabs/soci-snapshotter/blob/main/docs/index.md#zinfo). Here are some examples:

- The docs mention that zInfo is stored in binary format, instead of as human-readable JSON, but they don't specify which binary format is used. Judging from the source code, it looks like the binary format uses FlatBuffers. Maybe it'd be useful to document this explicitly.
- The docs describe checkpoints / compression info, but don't explain how it is calculated. Is it tightly coupled to this snapshotter's implementation of gzip?

**Comment by aochagavia** [comment]
In case anyone stumbles upon this in the future, I discovered a Rust crate that implements the ztoc index: [ztoc-rs](https://github.com/replit/ztoc-rs). It looks like it's [being used by replit](https://blog.replit.com/deployments-image-streaming) in production, which is nice.

**Comment by austinvazquez** [comment]
Hi, @aochagavia thanks for the interest in SOCI. I drafted #1125 to capture some of the feedback you provided to improve the documentation here.
